src/plugins/dashboard/public/application/actions/filters_notification_badge.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import { CoreStart, OverlayStart } from '@kbn/core/public';
+import { DataView } from '@kbn/data-views-plugin/public';
+import { isFilterableEmbeddable } from '@kbn/embeddable-plugin/public';
+import { createKibanaReactContext } from '@kbn/kibana-react-plugin/public';
+
+import { Action, IncompatibleActionError } from '../../services/ui_actions';
+
+import { toMountPoint } from '../../services/kibana_react';
+import { IEmbeddable, isErrorEmbeddable } from '../../services/embeddable';
+
+import { FiltersNotificationModal } from './filters_notification_modal';
+import { DashboardContainer } from '../embeddable';
+import { dashboardFilterNotificationBadge } from '../../dashboard_strings';
+
+export const BADGE_FILTERS_NOTIFICATION = 'ACTION_FILTERS_NOTIFICATION';
+
+export interface FiltersNotificationActionContext {
+  embeddable: IEmbeddable;
+}
+
+export class FiltersNotificationBadge implements Action<FiltersNotificationActionContext> {
+  public readonly id = BADGE_FILTERS_NOTIFICATION;
+  public readonly type = BADGE_FILTERS_NOTIFICATION;
+  public readonly order = 2;
+
+  private displayName = dashboardFilterNotificationBadge.getDisplayName();
+  private icon = 'filter';
+
+  constructor(
+    private theme: CoreStart['theme'],
+    private overlays: OverlayStart,
+    private uiSettings: CoreStart['uiSettings']
+  ) {}
+
+  public getDisplayName({ embeddable }: FiltersNotificationActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return this.displayName;
+  }
+
+  public getIconType({ embeddable }: FiltersNotificationActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return this.icon;
+  }
+
+  public isCompatible = async ({ embeddable }: FiltersNotificationActionContext) => {
+    return (
+      !isErrorEmbeddable(embeddable) &&
+      embeddable.getRoot().isContainer &&
+      isFilterableEmbeddable(embeddable) &&
+      embeddable.getFilters().length > 0
+    );
+  };
+
+  public execute = async (context: FiltersNotificationActionContext) => {
+    const { embeddable } = context;
+    const isCompatible = await this.isCompatible({ embeddable });
+    if (!isCompatible || !isFilterableEmbeddable(embeddable)) {
+      throw new IncompatibleActionError();
+    }
+
+    const { Provider: KibanaReactContextProvider } = createKibanaReactContext({
+      uiSettings: this.uiSettings,
+    });
```

Comment:
This is necessary for line 36 of `filter_items.tsx`:

https://github.com/elastic/kibana/blob/76d317be7e957f4fda1529dfb0f4f47c70c07b3e/src/plugins/unified_search/public/filter_bar/filter_item/filter_items.tsx#L33-L36

Otherwise, `kibana.services` is undefined for anything that happens inside of this modal.